### PR TITLE
Add `.readthedocs.yml` to use the same document dependencies on the CI and Read the Docs

### PR DESCRIPTION
**Pull request by harupy**
## Motivation

Fix https://github.com/optuna/optuna/issues/1353

## Description of the changes

- Remove `doc/requirements.txt` because it's no longer necessary.
- Add `.readthedocs.yml` to run `pip install .[document]` on Read the docs.
- Fix the version specifier for `sphinx` in `setup.py`.

---

I verified this approach works on my forked repo.

### Branch

https://github.com/harupy/optuna/tree/doc-test

### Documentation build log

https://readthedocs.org/projects/harupy-optuna/builds/11232421/

### Generated documentation

https://harupy-optuna.readthedocs.io/en/latest/reference/visualization.html#optuna.visualization.plot_param_importances

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def get_extras_require() -> Dict[str, List[str]]:
             "mlflow",
         ],
         # TODO(hvy): Unpin `sphinx` version after https://github.com/sphinx-doc/sphinx/issues/7807.
-        "document": ["sphinx<3.1.0", "sphinx_rtd_theme"],
+        "document": ["sphinx>=3.0.0,!=3.1.0", "sphinx_rtd_theme"],
```
Comment (harupy) [inline setup.py:64]: Read the Docs pre-installs `sphinx < 2` (which satisfies `< 3.1.0`) before running `pip install .[document]`, which prevents `pip install .[docuemnt]` from updating the sphinx version.

**Comment by hvy** [comment]
This sounds like a reasonable shift. Going through RTD's materials, it seems `.readthedocs.yml` is going to be the preferred way of specifying RTD build configurations. It should be noted that it's still in beta according to https://docs.readthedocs.io/en/stable/development/design/yaml-file.html, but changes are local and it shouldn't be too difficult to revert them in the worst case.

**Comment by harupy** [comment]
The build succeeded!

https://readthedocs.org/projects/optuna/builds/11239513/

The doc shows correct signatures now.

Screen Shot 2020-06-12 at 13 44 08

https://optuna.readthedocs.io/en/latest/reference/visualization.html

**Comment by hvy** [comment]
Awesome 👌

**Pull request merged**